ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/shared/header/header.component.spec.ts

```diff
@@ -1,64 +1,58 @@
+import { UserProfile } from 'src/app/modules/shared/models/user-profile.model';
+import { ProfileService } from 'src/app/services/profile.service';
+import { HeaderComponent } from './header.component';
 
 export class MockProfileService {
-
-  getUserProfile(): Promise<UserProfile> {
-      const profile = new UserProfile();
-      profile.role = 'professional';
-      profile.email = 'professional@hearings.hmcts.net';
-      return Promise.resolve(profile);
-  }
-
-  public get isLoggedIn(): boolean {
-      return true;
-  }
+    getUserProfile(): Promise<UserProfile> {
+        const profile = new UserProfile();
+        profile.role = 'professional';
+        profile.email = 'professional@hmcts.net';
+        return Promise.resolve(profile);
+    }
+
+    public get isLoggedIn(): boolean {
+        return true;
+    }
 }
 
-
-
-describe('HeaderComponent',
-  () => {
+describe('HeaderComponent', () => {
     let component: HeaderComponent;
     let fixture: ComponentFixture<HeaderComponent>;
 
-    const router = {
-      navigate: jasmine.createSpy('navigate')
-    };
-
     const profileService = new MockProfileService();
 
-    beforeEach(waitForAsync(() => {
-      TestBed.configureTestingModule({
-        declarations: [HeaderComponent],
-        providers: [
-          { provide: Router, useValue: router },
-          { provide: ProfileService, useValue: profileService },
-        ]
-      },
-      )
-        .compileComponents();
-    }));
+    beforeEach(
+        waitForAsync(() => {
+            TestBed.configureTestingModule({
+                imports: [RouterTestingModule],
+                declarations: [HeaderComponent],
+                providers: [{ provide: ProfileService, useValue: profileService }]
+            }).compileComponents();
+        })
+    );
 
     beforeEach(() => {
-      fixture = TestBed.createComponent(HeaderComponent);
-      component = fixture.componentInstance;
-      fixture.detectChanges();
+        fixture = TestBed.createComponent(HeaderComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
     });
 
     it('should navigate to logout page on pressing logout', () => {
-      component.loggedIn = true;
-      fixture.detectChanges();
+        component.loggedIn = true;
+        fixture.detectChanges();
+
+        const href = fixture.debugElement.query(By.css('#header-logout-link')).nativeElement.getAttribute('href');
+        expect(href).toEqual(`/${component.logoutRoute}`);
+    });
 
-      const link = fixture.debugElement.query(By.css('#header-logout-link'));
-      link.triggerEventHandler('click', null);
+    it('logout button is not rendered when user is not logged in', () => {
+        component.loggedIn = false;
+        fixture.detectChanges();
 
-      expect(router.navigate).toHaveBeenCalledWith(['/logout']);
+        const logoutButton = fixture.debugElement.query(By.css('#header-logout-link'));
+        expect(logoutButton).toBeNull();
```

Comment:
Add a test that checks it is present when logged in is true.

Reply:
Hi Farron, that edge case is covered by the other test

Reply:
I see now. I think it would be best if you reverted the test for the click event to how it was before, rather than testing the href and have a test for the logout button being present just to be more specific.

Reply:
I don't think it would be possible as router is no longer being injected through the constructor so I cant pass a mocked router and then check if its being called, initially thats what I was trying to do, but the problem here is that the router link is only referenced in the html template file unless I am missing something like a RouterTestingModule library for html templates?

Reply:
Ah yeah, makes sense. I didn't consider router being removed. Probably could use RouterTestingModule to be completely thorough but that's fine.